File: TextEditor/MyTextEditor/EditorMemento.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyTextEditor
{
    public class EditorMemento
    {
        public string Content { get; }
        public int CursorPosition { get; }
        public List<string> Selections { get; }

        public EditorMemento(string content, int cursorPosition, List<string> selections)
        {
            Content = content;
            CursorPosition = cursorPosition;
            Selections = selections;
        }
    }
}

File: Lab-12/TextEditorProject/src/EditorMemento.java
import java.util.List;

public class EditorMemento {
  private String content;
  private int cursorPosition;
  private List<String> selections;
  public EditorMemento(String content,int cursorPosition,List<String> selections){
      this.content = content;
      this.cursorPosition = cursorPosition;
      this.selections = selections;
  }

    public int getCursorPosition() {
        return cursorPosition;
    }

    public String getContent() {
        return content;
    }

    public List<String> getSelections() {
        return selections;
    }
}